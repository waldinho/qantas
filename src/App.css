.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.App-header {
  min-height: 10vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  background: #fff;
  box-shadow: 0 0.125rem 0.125rem 0 rgba(0,0,0,0.1);
  color: #323232;
  position: fixed;
  top: 0;
  width: -webkit-fill-available;
  z-index: 2;
}

button {
  display: inline-block;
  border-radius: .1875rem;
  font-weight: bold;
  font-size: 1rem;
  line-height: 1.5rem;
  text-align: center;
  text-transform: uppercase;
  text-decoration: none;
  padding: .688rem 1.5rem;
  color: #e40000;
  border: 2px solid #e40000;
  margin: 1rem 0 1.5rem 0;
}

button:hover {
  background: #e40000;
  color: #fff;
}

input {
  border: 2px solid #dadada;
  box-sizing: border-box;
  color: #323232;
  font-size: 18px;
  height: 48px;
  line-height: 25.5px;
  padding: 10.5px 35px 12px 14px;
  font-family: 'Abel', Arial, Verdana, sans-serif;
}

.search {
  position: fixed;
  top: 10vh;
  width: -webkit-fill-available;
  background: #323232;
  padding: 1vh;
  z-index: 1;
}