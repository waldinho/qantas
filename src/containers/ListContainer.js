import React, {useState} from 'react';
import styled from 'styled-components';

import List from '../components/List'
import DetailContainer from '../containers/DetailContainer'

const Wrapper = styled.div`

`;

const ListContainer = ({content}) => {
    const [active, setActive] = useState(true);
    console.log('List:', content)
    return(
        <>
            {content.slice(0, 21).map((item, i)=>{
                    return(
                        <Wrapper key={item.airportName}>
                        { active ?
                            <List 
                                airportCode={item.airportCode}
                                airportName={item.airportName}
                                domestic={item.domesticAirport}
                                international={item.internationalAirport}
                                cityName={item.city.cityName}
                                cityCode={item.city.cityCode}
                                countryName={item.country.countryCode}
                                countryCode={item.country.countryCode}
                                location={item.location}
                                region={item.region}
                        />  :
                            <DetailContainer 
                                airportCode={item.airportCode}
                                airportName={item.airportName}
                                domestic={item.domesticAirport}
                                international={item.internationalAirport}
                                cityName={item.city.cityName}
                                cityCode={item.city.cityCode}
                                countryName={item.country.countryCode}
                                countryCode={item.country.countryCode}
                                location={item.location}
                                region={item.region}
                            />
                        }
                        </Wrapper>
                    );                
                })
            }
        </>
    );
}

export default ListContainer;