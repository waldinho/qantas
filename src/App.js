  
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import './App.css';
import styled from 'styled-components'

import AirportContainer from "./containers/AirportContainer";

const AppWrapper = styled.div`
  text-align: center;
`
const App = () => (
 <Router>
    <>
      <AppWrapper>
        <AirportContainer />
      </AppWrapper>
    </>
 </Router>
);

export default App;
