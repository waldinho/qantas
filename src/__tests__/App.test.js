import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow } from 'enzyme';

import App from '../App';
import ListContainer from '../containers/ListContainer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

const dummyObj = [
  {
    airportCode:"AAA",
    internationalAirport:false,
    domesticAirport:false,
    regionalAirport:false,
    onlineIndicator:false,
    eticketableAirport:false,
    location:
      {
        aboveSeaLevel:-99999,
        latitude:17.25,
        latitudeRadius:-0.3040,
        longitude:145.30,longitudeRadius:-2.5395,
        latitudeDirection:"S",
        longitudeDirection:"W"
      },
    airportName:"Anaa",
    city:
      {
        cityCode:"AAA",
        cityName:"Anaa",
        timeZoneName:"Pacific/Tahiti"},
        country:{
          countryCode:"PF",
          countryName:"French Polynesia"
        },
        region:{
          regionCode:"SP",
          regionName:"South Pacific"
        }
      }
]

describe('<ListContainer /> with props', () => {
  const container = shallow(
    <Router>
      <ListContainer 
      content={dummyObj} />
    </Router>
  );
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  })
})
