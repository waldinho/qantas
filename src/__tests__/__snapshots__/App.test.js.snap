// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailContainer /> with props should match the snapshot 1`] = `"<div class=\\"sc-ifAKCX jtdfU\\"><div class=\\"sc-bxivhb hnjaHi\\"><div class=\\"main\\"><div><h2> - </h2><p>, , </p></div><p aria-label=\\"View Details\\"><a class=\\"link\\" href=\\"/\\">&gt;Back</a></p></div><div>Loading...</div></div></div>"`;

exports[`<ListContainer /> with props should match the snapshot 1`] = `"<input type=\\"text\\" value=\\"\\" placeholder=\\"Search airports...\\"/><div class=\\"sc-bwzfXH bFHdpU\\"><div class=\\"sc-bdVaJa lgVskm\\"><div class=\\"content__left\\"><div><h3 aria-label=\\"Anaa\\">AAA - Anaa</h3><p aria-label=\\"Anaa\\">Anaa, French Polynesia, South Pacific</p></div></div><div class=\\"content__right\\"><div><p aria-label=\\"View Details\\"><a class=\\"link\\" href=\\"/anaa\\">View Details</a></p></div></div></div></div>}"`;
